src/hybrid_reader.c

```diff
@@ -287,16 +317,22 @@ IndexIterator *NewHybridVectorIterator(VecSimIndex *index, char *score_field, KN
   hi->base.isValid = 1;
   hi->list = NULL;
   hi->iter = NULL;
+  hi->numIterations = 0;
 
   if (child_it == NULL) {
-    hi->mode = STANDARD_KNN;
-//  } else if (UseBF(child_it->NumEstimated(child_it), query, index)) {
-//    hi->mode = HYBRID_ADHOC_BF;
+    hi->mode = VECSIM_STANDARD_KNN;
   } else {
-    hi->mode = HYBRID_BATCHES;
-    //Todo: apply heuristics (batch_size = k / (vec_index_size*child_it->NumEstimated(child_it)))
+    size_t subset_size = child_it->NumEstimated(child_it->ctx);
+    if (subset_size > VecSimIndex_IndexSize(index)) {
```

Comment:
when could it be?

Reply:
It happens in union for example. This isn't accurate, it is an upper bound.

Reply:
Please leave a comment with the exlanation